Alicerce/Sources/Logging/Log.swift

```diff
@@ -0,0 +1,132 @@
+//
+//  Alicerce.swift
+//  Alicerce
+//
+//  Created by Meik Schutz on 03/04/17.
+//  Copyright © 2017 Mindera. All rights reserved.
+//
+
+import Foundation
+
+public final class Log {
+
+    private static var providers = [LogProvider]()
+
+    public static let defaultLevel: Level = Level.error
+
+    // MARK:- Provider Management
+
+    internal static var providerCount: Int {
+        return providers.count
+    }
+
+    public class func register(_ provider: LogProvider) {
+        let matchingProviders = providers.filter { (registeredProvider) -> Bool in
```

Comment:
I would suggest to use `contains` with closure or `lazy.filter` at least instead of filter :)